test/UglifyJsPlugin.test.js

```diff
@@ -42,6 +44,62 @@ describe('UglifyJsPlugin', () => {
     });
   });
 
+  it('should regenerate hash', () => {
+    const originalMainTemplateUpdateHashForChunk =
+      MainTemplate.prototype.updateHashForChunk;
+    const originalChunkTemplateUpdateHashForChunk =
+      ChunkTemplate.prototype.updateHashForChunk;
+    const mockMainTemplateUpdateHashForChunk = jest.fn();
+    const mockChunkTemplateUpdateHashFocChunk = jest.fn();
+
+    MainTemplate.prototype.updateHashForChunk = mockMainTemplateUpdateHashForChunk;
+    ChunkTemplate.prototype.updateHashForChunk = mockChunkTemplateUpdateHashFocChunk;
+
+    const compiler = createCompiler({
+      entry: {
+        js: `${__dirname}/fixtures/entry.js`,
+        mjs: `${__dirname}/fixtures/entry.mjs`,
+        importExport: `${__dirname}/fixtures/import-export/entry.js`,
+        AsyncImportExport: `${__dirname}/fixtures/async-import-export/entry.js`,
+      },
+      output: {
+        path: `${__dirname}/dist`,
+        filename: '[name].[contenthash].js',
+        chunkFilename: '[id].[name].[contenthash].js',
+      },
+    });
+
+    new UglifyJsPlugin().apply(compiler);
+
+    return compile(compiler).then((stats) => {
+      const errors = stats.compilation.errors.map(cleanErrorStack);
+      const warnings = stats.compilation.warnings.map(cleanErrorStack);
+
+      expect(errors).toMatchSnapshot('errors');
+      expect(warnings).toMatchSnapshot('warnings');
+
+      // On each chunk we have 2 calls (we have 1 async chunk and 4 initial).
+      // First call do `webpack`.
+      // Second call do `TerserPlugin`.
+
+      // We have 1 async chunk (1 * 2 = 2 calls for ChunkTemplate)
+      expect(mockMainTemplateUpdateHashForChunk).toHaveBeenCalledTimes(8);
+      // We have 4 initial chunks (4 * 2 = 8 calls for MainTemplate)
+      expect(mockChunkTemplateUpdateHashFocChunk).toHaveBeenCalledTimes(2);
```

Comment:
Are the comments and codes in wrong order?

Reply:
@hsxfjames oh, yep, feel free to send a PR with fix